nav2_recoveries/src/back_up.cpp

```diff
@@ -76,19 +77,46 @@ Status BackUp::onCycleUpdate()
   command_x_ < 0 ? cmd_vel.linear.x = -0.025 : cmd_vel.linear.x = 0.025;
 
   geometry_msgs::msg::Pose2D pose2d;
-  pose2d.x = current_pose.position.x + cmd_vel.linear.x * (1 / cycle_frequency_);
+  pose2d.x = current_pose.position.x;
   pose2d.y = current_pose.position.y;
   pose2d.theta = tf2::getYaw(current_pose.orientation);
 
-  if (!collision_checker_->isCollisionFree(pose2d)) {
-    stopRobot();
-    RCLCPP_WARN(node_->get_logger(), "Collision Ahead - Exiting BackUp");
-    return Status::SUCCEEDED;
+  if (!isCollisionFree(distance, cmd_vel, pose2d)) {
+    return Status::FAILED;
```

Comment:
Due to the current design of the behavior tree, we need to return `Status::SUCCEEDED` even if the recovery was not completed per #899.

Reply:
Also, I'd personally prefer to call stop the robot here rathen than inside `isCollisionFree`.

Reply:
Got it, I was torn between the `stopRobot` here or there, but eventually decided to hide it, but no problem I can move back out :+1:

Reply:
Commit to follow